include/z64cutscene.h

```diff
@@ -344,6 +344,42 @@ typedef enum CutsceneDestination {
     /* 0x77 */ CS_DEST_ZELDAS_COURTYARD_RECEIVE_LETTER
 } CutsceneDestination;
 
+/*
+correlation with `enum SceneLayer`:
+  CS_INDEX_MANUAL:     [SCENE_LAYER_CHILD_DAY .. SCENE_LAYER_ADULT_NIGHT]
+  CS_INDEX_SCRIPTED_*: SCENE_LAYER_CUTSCENE_FIRST
+
+correlation with `enum EntranceIndex`
+  CS_INDEX_MANUAL:     base entrance index + scene layer
+  CS_INDEX_SCRIPTED_*: base entrance index + SCENE_LAYER_CUTSCENE_FIRST + (cutscene index & 0xF)
+
+`z_select.c` naming:
+  CS_INDEX_NONE:       "Stage: night"
+  CS_INDEX_LAST:       "Stage: day"
+  CS_INDEX_SCRIPTED_*: "Stage demo 0*", between [0xFFF0 .. 0xFFFA]
+*/
+typedef enum CutsceneIndex {
+    CS_INDEX_NONE = 0x0000,
+    CS_INDEX_MANUAL = 0xFFEF,
+    CS_INDEX_SCRIPTED_0 = 0xFFF0,
+    CS_INDEX_SCRIPTED_1 = 0xFFF1,
+    CS_INDEX_SCRIPTED_2 = 0xFFF2,
+    CS_INDEX_SCRIPTED_3 = 0xFFF3,
+    CS_INDEX_SCRIPTED_4 = 0xFFF4,
+    CS_INDEX_SCRIPTED_5 = 0xFFF5,
+    CS_INDEX_SCRIPTED_6 = 0xFFF6,
+    CS_INDEX_SCRIPTED_7 = 0xFFF7,
+    CS_INDEX_SCRIPTED_8 = 0xFFF8,
+    CS_INDEX_SCRIPTED_9 = 0xFFF9,
+    CS_INDEX_SCRIPTED_A = 0xFFFA,
+    CS_INDEX_SCRIPTED_B = 0xFFFB,
+    CS_INDEX_SCRIPTED_C = 0xFFFC,
+    CS_INDEX_SCRIPTED_D = 0xFFFD,
+    CS_INDEX_SCRIPTED_E = 0xFFFE,
+    CS_INDEX_SCRIPTED_F = 0xFFFF,
+    CS_INDEX_LAST = 0x8000,
```

Comment:
Also don't understand why this is called `CS_INDEX_LAST`, and why it is placed all the way down here.

Reply:
tokinoma_scene actually only has layers 0-14, so the max cutscene index that loads a valid layer is actually 0xFFFA, which is what z_select goes up to. `ENTR_TEMPLE_OF_TIME_0_15` is unreachable without glitches.

The cutscene count must start at 0, not 4:

* It makes it easier to logically follow what `SCENE_LAYER_CUTSCENE_FIRST + (gSaveContext.save.cutsceneIndex & 0xF);` is accomplishing. 

* In MM, the game was refactored to no longer need separate layers to spawn actors at different times of day, so now only layer 0 is a default playable layer, and layers 1+ become the cutscene layers.

Reply:
https://github.com/zeldaret/oot/pull/2286#discussion_r1861141789

um, `z_select.c` is a debug menu, so I'd rather not. also, you see, `CS_INDEX_AUTO` means "choose `gSaveContext.sceneLayer` based on age and day time", so a `_NONE_NIGHT` wouldn't fit here completely  

I believe, the devs simply used `0x0000` and `0x8000` there as spare values

Reply:
Based on the original name of (`day_time`), and the fact that 0x0000 and 0x8000 are used for day and night respectively in map select (which probably was made early in development), I think `cutsceneIndex` was actually intended to be a used for time-of-day first, and later 0xFFFx were the ones used as "spare values".

Don't both 0x0000 and 0x8000 mean "choose `gSaveContext.sceneLayer` based on age and day time"? And even if you discount debug menus, value 0x8000 is seemingly used for `CS_DEST_DEATH_MOUNTAIN_TRAIL`. I'm not in love with something like `CS_INDEX_NIGHT` and `CS_INDEX_DAY` for these values, but I think it's an improvement over the current names so I wanted to throw it out there

Reply:
hmm, that's an interesting assumption. ok, i'm not entirely against renaming those defines, but want to show some other observations:

1) it's true that here anything less than `0xFFF0` will be considered "age and day time"
https://github.com/zeldaret/oot/blob/6efea0a85fdfb44b44db6a9f456789212147f505/src/code/z_play.c#L349
but nothing (about DMT later) game-wise, i.e. besides debug menu, in the code uses `0x8000` for this purpose

2) indeed, there's one case for DMT
https://github.com/zeldaret/oot/blob/6efea0a85fdfb44b44db6a9f456789212147f505/src/code/z_demo.c#L749
but in reality it's an unused destination
https://github.com/zeldaret/oot/blob/6efea0a85fdfb44b44db6a9f456789212147f505/include/z64cutscene.h#L244
which renders it as unreliable source of information

3) even if we rely on the debug menu for naming it
`0x0000` is "night" (utilised everywhere)
https://github.com/zeldaret/oot/blob/6efea0a85fdfb44b44db6a9f456789212147f505/src/overlays/gamestates/ovl_select/z_select.c#L559-L561
`0x8000` is "day" (set in one unused case)
https://github.com/zeldaret/oot/blob/6efea0a85fdfb44b44db6a9f456789212147f505/src/overlays/gamestates/ovl_select/z_select.c#L563-L565
seems a bit misleadeing to name a "child or adult, day or night" just "night"

because you're somewhat 240 PRs more experienced and familiar with the code than me I can give up and accept your suggestion. but, yeah, am not enthusiastic about `CS_INDEX_NIGHT` / `CS_INDEX_DAY` either

maybe we can name it `CS_INDEX_DEFAULT` / `CS_INDEX_DEFAULT_UNUSED` if that sounds better?